### ISSUE TITLE
Upgrade to the newest bondagejs

### ISSUE THREAD

**Issue by blurymind** [4ian/GDevelop#6451]
## Description

The current version of bondagejs which gdevelop uses doesnt have full yarn 2 syntax compliance.
Things like inline expressions dont work, so for example you cant show variables in text
` The player name is ${playerName}`
## Solution suggested
 
There is  a branch of bondagejs that upgrades it to the yarn2 spec
https://github.com/hylyh/bondage.js/tree/2.0

in theory all thats needed is to simply swap out bondagejs with the new one and test it in gdevelop

**Comment by blurymind** [comment]
btw one added extra of this branch is that the library's deployment size has considerably been shrunk

**Comment by 4ian** [comment]
That seems like a great idea! As usual I would say the most important is to triple check any potential breaking changes/regressions.

**Comment by blurymind** [comment]
yes ofcourse, it may be a case that the wrapper i wrote for it might need some subtle tweaks. It will need to be tested with all the example projects and work exactly the same on them

**Comment by blurymind** [comment]
another idea I have is to create a module api and a wasm target of the rust port of yarn spinner.

That would probably be even better as it will give us a much better interpreter than bondagejs, which is in a much healthier condition. I created a ticket with intent here:
https://github.com/YarnSpinnerTool/YarnSpinner-Rust/issues/213

**Comment by janhohenheim** [comment]
Chiming in to say that I'm very happy to help out if you have any questions regarding the Rust port :)
